Index trajectories

danielag49 opened the pull request:
this is the index of trajectories using city_name and includes that only gonna show the name of the cities

danielag49 committed:
this is the index of trajectories using city_name

dsalazar357 commented:
Hey there. May we change the name of the folder to `trajectories` instead of `trayectories`, don't you agree?
Sincere thanks c:

Also, why is the instance variable called `@TrajectoriesController`, it might sound a tad reiterative perhaps.
I reckon my suggestion would be to just name it `@trajectories`, since it's going to be iterated in the view, if I assume right.

danielag49 committed:
we change all the names the trayectory to trajectory and also change the coments that carlos said

danielag49 committed:
we corrected the names inside the migrations, from trayectories to trajectory, we also made the changes in the index and corrected the stops field in the migration, we also deleted the database, recreated it and migrated it so it won't have any problems :)

danielag49 commented:
we corrected the names inside the migrations, from trayectories to trajectory, we also made the changes in the index and corrected the stops field in the migration, we also deleted the database, recreated it and migrated it so it won't have any problems :)

dsalazar357 commented:
Looks good to me c:

mtopete1996 commented on app/controllers/trajectories_controller.rb:
```diff
@@ -0,0 +1,5 @@
+class TrajectoriesController < ApplicationController
+  def index
+    @trajectories = trajectories.includes(:city).all
```
where does trajectories.includes comes from?

danielag49 replied:
done

danielag49 committed:
I just change the name trajectories to Trajectory because that wasn't the name of the class i'm using

danielag49 committed:
i change trajectories.includes to Tryectory.includes

danielag49 committed:
This is the index about trajectories that is going to show the cities that have trajectories

mtopete1996 reviewed (changes requested):
You're missing the routes

danielag49 committed:
just add the resources and put a space in params

jesusMava commented on db/migrate/20240507232734_create_schedules.rb:
```diff
@@ -2,7 +2,7 @@ class CreateSchedules < ActiveRecord::Migration[7.1]
     create_table :schedules do |t|
       t.time :time
-      t.references :trayectories, null: false, foreign_key: true
+      t.references :trajectory, null: false, foreign_key: true
```
@REYNALopez12 Has the same change 🤔

jesusMava commented on app/controllers/trajectories_controller.rb:
```diff
@@ -0,0 +1,4 @@
+class TrajectoriesController < ApplicationController
+  def index 
+    Trajectory.where(city_id: params.fetch(:city_id,[]))
```
🤔 how does your view handle this? and why a fetch? 
what if there is no trajectory?

danielag49 replied:
help @mtopete1996  help

mtopete1996 replied:
```suggestion
    Trajectory.where(city_id: params[:city_id])
```

This would work the same way

jesusMava commented on db/migrate/20240419210053_create_stops.rb:
```diff
@@ -2,7 +2,7 @@ class CreateStops < ActiveRecord::Migration[7.1]
     create_table :stops do |t|
       t.string :name, null: false
-      t.column :trayectory_geom, :geometry, limit: "POINT"
+      t.column :trajectory_geom, :geometry, limit: "POINT"
```
nice, you fix all of these typos... 👍

danielag49 committed:
I removed the fetch method and just let the code look for records in the Trajectory table where the city_id field matches the city_id parameter

danielag49 committed:
Merge branch 'main' into indexTrajectories
